File: src/test/java/org/base/leetcode/interview75/graphs/bfs/GraphBfsTests.java
package org.base.leetcode.interview75.graphs.bfs;

import org.junit.jupiter.api.Test;

import static org.base.leetcode.util.Helper.toNestedCharArray;
import static org.base.leetcode.util.Helper.toNestedIntArray;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class GraphBfsTests {
    @Test
    public void testNearestExit() {
        NearestExit exit = new NearestExit();
        assertEquals(1, exit.nearestExit(
                toNestedCharArray("[[\"+\",\"+\",\".\",\"+\"],[\".\",\".\",\".\",\"+\"],[\"+\",\"+\",\"+\",\".\"]]"),
                new int[] {1,2}));
        assertEquals(2, exit.nearestExit(
                toNestedCharArray("[[\"+\",\"+\",\"+\"],[\".\",\".\",\".\"],[\"+\",\"+\",\"+\"]]"),
                new int[] {1,0}));
        assertEquals(-1, exit.nearestExit(
                toNestedCharArray("[[\".\",\"+\"]]"),
                new int[] {0,0}));
        assertEquals(12, exit.nearestExit(
                toNestedCharArray("[[\"+\",\".\",\"+\",\"+\",\"+\",\"+\",\"+\"],[\"+\",\".\",\"+\",\".\",\".\",\".\",\"+\"],[\"+\",\".\",\"+\",\".\",\"+\",\".\",\"+\"],[\"+\",\".\",\".\",\".\",\"+\",\".\",\"+\"],[\"+\",\"+\",\"+\",\"+\",\"+\",\".\",\"+\"]]"),
                new int[] {0,1}));
    }

    @Test
    public void testRottingOranges() {
        RottingOranges rotting = new RottingOranges();
        assertEquals(4, rotting.orangesRotting(
                toNestedIntArray("[[2,1,1],[1,1,0],[0,1,1]]")));
        assertEquals(-1, rotting.orangesRotting(
                toNestedIntArray("[[2,1,1],[0,1,1],[1,0,1]]")));
        assertEquals(0, rotting.orangesRotting(
                toNestedIntArray("[[0,2]]")));
    }
}

File: src/test/java/org/base/leetcode/interview75/graphs/bfs/GraphBfsTests.kt
package org.base.leetcode.interview75.graphs.bfs

import org.base.leetcode.util.Kelper.toNestedCharArray
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class GraphBfsTests {
    @Test
    fun `should find nearest exit`() {
        val exit = NearestExit()
        assertEquals(1, exit.nearestExit(
            toNestedCharArray("[[\"+\",\"+\",\".\",\"+\"],[\".\",\".\",\".\",\"+\"],[\"+\",\"+\",\"+\",\".\"]]"),
            intArrayOf(1,2)))
        assertEquals(2, exit.nearestExit(
            toNestedCharArray("[[\"+\",\"+\",\"+\"],[\".\",\".\",\".\"],[\"+\",\"+\",\"+\"]]"),
            intArrayOf(1,0)))
        assertEquals(-1, exit.nearestExit(
            toNestedCharArray("[[\".\",\"+\"]]"),
            intArrayOf(0,0)))
        assertEquals(12, exit.nearestExit(
            toNestedCharArray("[[\"+\",\".\",\"+\",\"+\",\"+\",\"+\",\"+\"],[\"+\",\".\",\"+\",\".\",\".\",\".\",\"+\"],[\"+\",\".\",\"+\",\".\",\"+\",\".\",\"+\"],[\"+\",\".\",\".\",\".\",\"+\",\".\",\"+\"],[\"+\",\"+\",\"+\",\"+\",\"+\",\".\",\"+\"]]"),
            intArrayOf(0,1)))
    }
}